### ISSUE TITLE
Class "Symfony\Component\Security\Http\Firewall\BasicAuthenticationListener" used for service "security.authentication.listener.basic.main" cannot be found.

### ISSUE THREAD

**Issue by alexander-schranz** [symfony/symfony#44147]
### Symfony version(s) affected

5.4.0-BETA-3

### Description

When I'm running my testa against Symfony 5.4.0-BETA-3 or current dev we are getting a strange error: 

The following package where installed:

```
Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: Class "Symfony\Component\Security\Http\Firewall\BasicAuthenticationListener" used for service "security.authentication.listener.basic.main" cannot be found.
```

### How to reproduce

https://github.com/massiveart/MassiveSearchBundle/runs/4266812846?check_suite_focus=true

Running on PHP 8.0 and install dev dependencies to run the test against that repository.

### Possible Solution

Maybe have todo that the security-bundle 5:4 allows 6.0 packages but tries to register a `BasicAuthenticationListener` which does not exist in 6.0 security package.

**Comment by derrabus** [comment]
> Maybe have todo that the security-bundle 5:4 allows 6.0 packages but tries to register a `BasicAuthenticationListener` which does not exist in 6.0 security package.

It only does that if you use the old authenticator system that has been removed in Symfony 6. We can improve the error message that you receive, but your tests will still fail as long as you don't migrate to the new system.

**Comment by wouterj** [comment]
@derrabus is correct and I do expect you would see a deprecation notice about this. Please consider updating the `SYMFONY_DEPRECATIONS_HELPER` env variable that the bundle is using to [something like `max[direct]=0`](https://symfony.com/doc/current/components/phpunit_bridge#direct-and-indirect-deprecations).

Also, I would recommend installing Symfony Flex in the bundle's CI workflow and using the `SYMFONY_REQUIRE` env var to limit Symfony to a specific version. See for instance [SymfonyAclBundle's github actions workflow](https://github.com/symfony/acl-bundle/blob/main/.github/workflows/ci.yml#L42-L45)

**Comment by alexander-schranz** [comment]
> It only does that if you use the old authenticator system that has been removed in Symfony 6. > We can improve the error message that you receive, but your tests will still fail as long as you don't migrate to the new system.

Make sense for me maybe a better error message would be great here. Maybe some check for `enable_authenticator_manager` is not set to true and clas `BasicAuthenticationListener` does exist. Tell the user to  use the new authenticator system or when using symfony 6 package. Or we could only allow symfony 5.4 package in combination with the 5.4 bundle, then this kind of dependencies could not be installed.

**Comment by derrabus** [comment]
In a way, the problem should be gone now because the Guard component (as a mandatory dependency of SecurityBundle) effectivly downgrades the secrity packages to 5.4 because of #44172. I'm not entirely happy with that, I must admit. 🤔

**Comment by alexander-schranz** [comment]
Tested and did work after I added a conflict to `5.4.0-BETA1` of the `symfony/security-guard` package. Thank you all for your fast responses and @derrabus thank you for the fix!

Closing as fixed in #44172
